src/images/imagePlot.component.tsx

```diff
@@ -0,0 +1,213 @@
+
+export interface ImagePlotProps {
+  data: CrosshairDimensionType['intensity'];
+  crosshairPosition: number;
+  image?: string;
+}
+
+const commonChartOptions: ChartOptions<'line'> = {
+  responsive: false,
+  maintainAspectRatio: false,
+  interaction: {
+    mode: 'index',
+    intersect: true,
+  },
+  plugins: {
+    legend: {
+      display: false,
+    },
+    annotation: { annotations: {} },
+  },
+};
+
+const YChartOptions: ChartOptions<'line'> = {
+  ...commonChartOptions,
+  indexAxis: 'y',
+  scales: {
+    y: {
+      type: 'linear',
+      // min max get set based on image size in useEffect
+      ticks: { padding: 0, align: 'start' },
+      reverse: true,
+    },
+    x: {
+      type: 'linear',
+      min: 0,
+      max: 255,
+      ticks: { padding: 0 },
+    },
+  },
+};
+
+const XChartOptions: ChartOptions<'line'> = {
+  ...commonChartOptions,
+  scales: {
+    y: {
+      type: 'linear',
+      ticks: { padding: 0 },
+      min: 0,
+      max: 255,
+    },
+    x: {
+      type: 'linear',
+      // min max get set based on image size in useEffect
+      ticks: { padding: 0, align: 'end' },
+    },
+  },
+};
+
+export const XImagePlot = (props: ImagePlotProps) => {
+  const { data, image, crosshairPosition } = props;
+  return (
+    <ImagePlot
+      data={data}
+      image={image}
+      crosshairPosition={crosshairPosition}
+      type="x"
+      chartOptions={XChartOptions}
+    />
+  );
+};
+
+export const YImagePlot = (props: ImagePlotProps) => {
+  const { data, image, crosshairPosition } = props;
+  return (
+    <ImagePlot
+      data={data}
+      image={image}
+      crosshairPosition={crosshairPosition}
+      type="y"
+      chartOptions={YChartOptions}
+    />
+  );
+};
+
+const ImagePlot = (
+  props: ImagePlotProps & {
+    type: 'x' | 'y';
+    chartOptions: ChartOptions<'line'>;
+  }
+) => {
+  const { data, crosshairPosition, image, type, chartOptions } = props;
+
+  const [imageElement, setImageElement] = React.useState<HTMLImageElement>();
+
+  // set the initial options
+  const [optionsString, setOptionsString] = React.useState(
+    JSON.stringify(chartOptions)
+  );
+
+  const [dataString, setDataString] = React.useState(
+    JSON.stringify({
+      labels: data.x,
+      datasets: [
+        {
+          data: data.y,
+          borderColor: 'red', // same colour as crosshairPosition
+          borderWidth: 1,
+          pointRadius: 0,
+          pointHitRadius: 2,
+        },
+      ],
+    } satisfies ChartData<'line'>)
+  );
+
+  const [resizeString, setResizeString] = React.useState(
+    JSON.stringify(
+      type === 'x'
+        ? {
+            width: imageElement?.naturalWidth
+              ? imageElement.naturalWidth + 29
+              : 200,
+            height: 200,
+          }
+        : {
+            width: 200,
+            height: imageElement?.naturalHeight
+              ? imageElement.naturalHeight + 22
+              : 200,
```

Comment:
Why does the natural width and height need 29 and 22 added ?

Reply:
Ah, I should add a comment for this, but I needed to add some padding to account for the width/height of the Chart.js axis ticks (aka the numbers) so that the actual axis line itself aligns with the image.

Maybe I can try and calculate it based on the default config variables in Chart.js - but they may just have to remain magic numbers...

I will at least extract both out to variables at the top of the file (to make them easy to edit if we need) and explain what they're for